### PR TITLE
nrf_security: Update to nrf_cc3xx 0.9.17 and nrf_oberon 3.0.13

### PR DESCRIPTION
**Pull request by joerchan**
Update crypto runtime libraries:
nrf_cc3xx 0.9.17
nrf_oberon 3.0.13

Compile Oberon crypto drivers from source.

### CODE REVIEW

**Review by Vge0rge — APPROVED** [review]

**Review by frkv — APPROVED** [review]
LGTM

File: crypto/nrf_cc310_platform/CHANGELOG.rst [hunk 1]
```diff
@@ -9,6 +9,34 @@ Changelog - nrf_cc3xx_platform
 
 All notable changes to this project are documented in this file.
 
+nrf_cc3xx_platform - 0.9.17
+***************************
+
+New version of the library with the following features:
+
+* Removed mbedtls_hardware_poll limitation on buffer size, it now accepts any buffer size.
```
Comment (melwee01) [inline crypto/nrf_cc310_platform/CHANGELOG.rst:17]: Same formatting note as above, but I'm not sure what this hardware_poll falls under.

```suggestion
* Removed mbedtls_hardware_poll limitation on buffer size. It now accepts any buffer size.
```

Should be two sentences. But what does 'it' refer to? The library?

File: crypto/nrf_cc310_platform/CHANGELOG.rst [hunk 2]
```diff
@@ -9,6 +9,34 @@ Changelog - nrf_cc3xx_platform
...
+Library built against Mbed TLS version 3.3.0.
+
+Added
+=====
+
+Added a new build of nrf_cc3xx_platform libraries for nRF9160, nRF52840, and nRF5340.
+
+.. note::
+
+   The *short-wchar* libraries are compiled with a wchar_t size of 16 bits.
```
Comment (melwee01) [inline crypto/nrf_cc310_platform/CHANGELOG.rst:29]: Appropriate formatting for wchar_t?

File: crypto/nrf_oberon/CHANGELOG.rst [hunk 3]
```diff
@@ -9,6 +9,120 @@ Changelog - nRF Oberon
...
 All notable changes to this project are documented in this file.
 
+nrf_oberon - 3.0.13
+*******************
+
+New version of the nrf_oberon library with the following changes.
+
+Added
+=====
+* Added Oberon PSA core, a heavily code-size optimized and efficient implementation of PSA core licensed for use on Nordic devices.
```
Comment (melwee01) [inline crypto/nrf_oberon/CHANGELOG.rst:19]:
```suggestion

* Added Oberon PSA core, a heavily code-size optimized and efficient implementation of PSA core licensed for use on Nordic Semiconductor devices.
```

File: crypto/nrf_oberon/CHANGELOG.rst [hunk 4]
```diff
@@ -9,6 +9,120 @@ Changelog - nRF Oberon
...
+  - HMAC, CMAC and CBC MAC
+  - RSA PKCS#1 1.5 and 2.1
+* Added custom Oberon PSA crypto drivers as source distribution:
+  - CTR_DRBG
+  - HMAC_DRBG
+  - SPAKE2+ using ECC curve type secp256r1
+  - SRP
+
+Removed
+=======
```
Comment (melwee01) [inline crypto/nrf_oberon/CHANGELOG.rst:44]:
```suggestion
=======

```

File: crypto/nrf_oberon/CHANGELOG.rst [hunk 5]
```diff
@@ -9,6 +9,120 @@ Changelog - nRF Oberon
...
+* Removed binary distribution of Oberon PSA crypto drivers
```
Comment (melwee01) [inline crypto/nrf_oberon/CHANGELOG.rst:45]:
```suggestion
* Removed binary distribution of Oberon PSA crypto drivers.
```

File: crypto/nrf_oberon/CHANGELOG.rst [hunk 6]
```diff
@@ -9,6 +9,120 @@ Changelog - nRF Oberon
...
+    * ``cortex-m33/soft-float/short-wchar/liboberon_mbedtls_3.0.13.a``
+
+  * Keil
+
+    * ``cortex-m33/hard-float/short-wchar/oberon_3.0.13.lib``
+    * ``cortex-m33/hard-float/short-wchar/oberon_mbedtls_3.0.13.lib``
+    * ``cortex-m33/soft-float/short-wchar/oberon_3.0.13.lib``
+    * ``cortex-m33/soft-float/short-wchar/oberon_mbedtls_3.0.13.lib``
+
+* nrf_oberon, nrf53 network core variants
```
Comment (melwee01) [inline crypto/nrf_oberon/CHANGELOG.rst:75]:
```suggestion
* nrf_oberon, nRF53 network core variants
```

**Review by melwee01 — CHANGES_REQUESTED** [review]

File: crypto/nrf_cc310_platform/CHANGELOG.rst (hunk 1 above)
Comment (joerchan) [reply crypto/nrf_cc310_platform/CHANGELOG.rst:17]: "It" is mbedtls_hardware_poll , which is a function.

File: crypto/nrf_cc310_platform/CHANGELOG.rst (hunk 2 above)
Comment (joerchan) [reply crypto/nrf_cc310_platform/CHANGELOG.rst:29]: \`\`wchar_t\`\`?

File: crypto/nrf_cc310_platform/CHANGELOG.rst (hunk 1 above)
Comment (joerchan) [reply crypto/nrf_cc310_platform/CHANGELOG.rst:17]: Removed limitation in ``mbedtls_hardware_poll`` function on buffer size. It now accepts any buffer size.

File: crypto/nrf_cc310_platform/CHANGELOG.rst (hunk 2 above)
Comment (joerchan) [reply crypto/nrf_cc310_platform/CHANGELOG.rst:29]: :c:type:t

File: crypto/nrf_oberon/CHANGELOG.rst (hunk 6 above)
Comment (joerchan) [reply crypto/nrf_oberon/CHANGELOG.rst:75]: I feel like this is just plain wrong.
As you can see from the paths, we have coretx-m specific builds, not SoC specific builds.
I would suggest changing to Cortex M33 and Cortex M4

File: crypto/nrf_oberon/CHANGELOG.rst [hunk 7]
```diff
@@ -9,6 +9,120 @@ Changelog - nRF Oberon
...
+  - SPAKE2+ using ECC curve type secp256r1
+  - SRP
+
+Removed
+=======
+* Removed binary distribution of Oberon PSA crypto drivers
+
+Library built against Mbed TLS version 3.3.0.
+
+Added the following Oberon crypto libraries for nRF9160, nRF53, nRF52, and nRF51 architectures.
```
Comment (joerchan) [inline crypto/nrf_oberon/CHANGELOG.rst:49]: Strangely enough, we are using nRF9160 i.e a specifig SOC, but nRF53, i.e a series.
This is a maintenance problem for adding new SoCs. for example the new nrf9161.

File: crypto/nrf_cc310_platform/CHANGELOG.rst (hunk 1 above)
Comment (melwee01) [reply crypto/nrf_cc310_platform/CHANGELOG.rst:17]: Somehow, when it's formatted, it's easier for me to understand that 'it' refers to the function. looks good now with that.

**Review by melwee01 — APPROVED** [review]
LGTM. I do wonder about how we'll refer to 91 series devices and the 9160 once the 9161 is out.

**Pull request merged**